src/Text/Pandoc/Readers/MediaWiki.hs

```diff
@@ -289,37 +296,48 @@ cellsep = try $ do
 
 tableCaption :: PandocMonad m => MWParser m Inlines
 tableCaption = try $ do
+  optional rowsep
   guardColumnOne
   skipSpaces
   sym "|+"
   optional (try $ parseAttrs *> skipSpaces *> char '|' *> blanklines)
   trimInlines . mconcat <$>
     many (notFollowedBy (cellsep <|> rowsep) *> inline)
 
-tableRow :: PandocMonad m => MWParser m [((Alignment, Double), Blocks)]
+tableRow :: PandocMonad m => MWParser m [(Double, Cell)]
 tableRow = try $ skipMany htmlComment *> many tableCell
 
-tableCell :: PandocMonad m => MWParser m ((Alignment, Double), Blocks)
+tableCell :: PandocMonad m => MWParser m (Double, Cell)
 tableCell = try $ do
   cellsep
   skipMany spaceChar
-  attrs <- option [] $ try $ parseAttrs <* skipSpaces <* char '|' <*
+  attribs <- option [] $ try $ parseAttrs <* skipSpaces <* char '|' <*
                                  notFollowedBy (char '|')
   skipMany spaceChar
   pos' <- getPosition
   ls <- T.concat <$> many (notFollowedBy (cellsep <|> rowsep <|> tableEnd) *>
                             ((snd <$> withRaw table) <|> countChar 1 anyChar))
   bs <- parseFromString (do setPosition pos'
                             mconcat <$> many block) ls
-  let align = case lookup "align" attrs of
+  let align = case lookup "align" attribs of
                     Just "left"   -> AlignLeft
                     Just "right"  -> AlignRight
                     Just "center" -> AlignCenter
                     _             -> AlignDefault
-  let width = case lookup "width" attrs of
+  let width = case lookup "width" attribs of
                     Just xs -> fromMaybe 0.0 $ parseWidth xs
                     Nothing -> 0.0
-  return ((align, width), bs)
+  let rowspan = RowSpan . fromMaybe 1 $
+                safeRead =<< lookup "rowspan" attribs
+  let colspan = ColSpan . fromMaybe 1 $
+                safeRead =<< lookup "colspan" attribs
+  let handledAttribs = ["align", "colspan", "rowspan"]
+      attribs' = foldr go [] attribs
```

Comment:
I had missed this the first time: I think this could be shortened to ``filter ((`notElem` handledAttribs) . fst) attribs``. Or, less compact but more readable, list comprehension:

``` haskell
attribs' = [ (k, v) | (k, v) <- attribs
                    , k `notElem` handledAttribs
           ]
```

I'll make a change after merging.

Reply:
Thanks. Acutally I referred to [HTML table reader](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Readers/HTML/Table.hs) here, and this is a simplied version to filter attributes.

Reply:
Ah, that makes sense :+1: